test/e2e/operator.go

```diff
@@ -379,36 +387,125 @@ var _ = Describe("ARO Operator - Azure Subnet Reconciler", func() {
 })
 
 var _ = Describe("ARO Operator - MUO Deployment", func() {
-	ctx := context.Background()
+	When("we have a fresh ARO cluster", func() {
+		It("must deploy MUO by default with FIPS crypto mandated", func() {
+			hasMUOFIPSCryptoMandatedFn := hasMUOFIPSCryptoMandatedHigherOrderFn(context.Background())
 
-	It("must be deployed by default with FIPS crypto mandated", func() {
-		muoIsDeployed := func() (bool, error) {
-			By("getting MUO pods")
-			pods, err := clients.Kubernetes.CoreV1().Pods("openshift-managed-upgrade-operator").List(ctx, metav1.ListOptions{
-				LabelSelector: "name=managed-upgrade-operator",
-			})
-			if err != nil {
-				return false, err
-			}
-			if len(pods.Items) != 1 {
-				return false, fmt.Errorf("%d managed-upgrade-operator pods found", len(pods.Items))
-			}
+			err := wait.PollImmediate(interval, duration, hasMUOFIPSCryptoMandatedFn)
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
 
-			By("getting logs from MUO")
-			b, err := clients.Kubernetes.CoreV1().Pods("openshift-managed-upgrade-operator").GetLogs(pods.Items[0].Name, &corev1.PodLogOptions{}).DoRaw(ctx)
-			if err != nil {
-				return false, err
-			}
+	When("MUO is deleted on the ARO cluster", func() {
+		It("should reconcile the MUO deployment", func() {
+			ctx := context.Background()
+			doesMUODeploymentExistFn := doesMUODeploymentExistHigherOrderFn(ctx)
 
-			By("verifying that MUO has FIPS crypto mandated by reading logs")
-			return strings.Contains(string(b), `msg="FIPS crypto mandated: true"`), nil
-		}
+			By("waiting for the MUO deployment to be ready")
+			err := wait.PollImmediate(interval, duration, doesMUODeploymentExistFn)
+			Expect(err).NotTo(HaveOccurred())
 
-		err := wait.PollImmediate(30*time.Second, 10*time.Minute, muoIsDeployed)
-		Expect(err).NotTo(HaveOccurred())
+			By("getting the MUO deployment to delete")
+			deploymentToDelete := getMUODeployment(ctx)
```

Comment:
do we need to get the deployment here or can we just call delete on the deployment name? We know it already because it uses a conventional name

Reply:
You're right, thanks! Removed that.